src/Blue/Tools/Api/Client.php

```diff
@@ -225,24 +218,33 @@ public function setLogger(LoggerInterface $logger)
     }
 
     /**
-     * Returns the specified request option or all options if none specified
+     * Returns the specified request option or all options if none specified.
+     *
      * @param null $keyOrPath
+     *
      * @return array|mixed|null
      */
     public function getRequestOption($keyOrPath = null)
     {
-        return $this->guzzleClient->getDefaultOption($keyOrPath);
+        return $this->guzzleClient->getConfig($keyOrPath);
     }
 
     /**
-     * Sets a request option for future requests
-     * @param $keyOrPath
+     * Sets a request option for future requests.
+     *
+     * @param $key
      * @param $value
+     *
      * @return $this
      */
-    public function setRequestOption($keyOrPath, $value)
+    public function setRequestOption($key, $value)
     {
-        $this->guzzleClient->setDefaultOption($keyOrPath, $value);
+        $config = array_merge(
+            $this->guzzleClient->getConfig(),
+            [$key => $value]
+        );
+
+        $this->guzzleClient = new GuzzleClient($config);
         return $this;
```

Comment:
Yes. The config array actually includes `['handler' => $handler]`, so this call copies that over to the new client object:

```php
$config = array_merge(
    $this->guzzleClient->getConfig(),
    [$key => $value]
);
```